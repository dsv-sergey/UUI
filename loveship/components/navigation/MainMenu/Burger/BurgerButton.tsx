import { withMods, IHasIcon, IDropdownToggler } from '@epam/uui';
import {  Button, ButtonProps } from '@epam/uui-components';
import * as svgTriangle from '../../../icons/triangle.svg';
import * as css from './BurgerButton.scss';

export interface BurgerButtonMods extends IHasIcon, IDropdownToggler {
    type?: 'primary' | 'secondary';
    indentLevel?: number;
}

export const BurgerButton = withMods<ButtonProps, BurgerButtonMods>(
    Button,
    props => [
        css.root,
        css['button-' + (props.type || 'primary')],
        css['indent-' + (props.indentLevel || 0)],
        props.isDropdown && css.dropdown,
        props.icon && css.hasIcon,
    ],
    () => ({ dropdownIcon: svgTriangle, dropdownIconPosition: 'left' }),
);
