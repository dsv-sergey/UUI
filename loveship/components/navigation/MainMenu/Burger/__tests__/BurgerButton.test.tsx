import React from "react";
import renderer from "react-test-renderer";
import {BurgerButton} from "../BurgerButton";
import * as acceptIcon from "../../../../icons/accept-12.svg";

describe("BurgerButton", () => {
    it("should be rendered correctly", () => {
        const tree = renderer
            .create(<BurgerButton/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should be rendered correctly with props', () => {
        const tree = renderer
            .create(<BurgerButton
                caption='Test button'
                icon={ acceptIcon }
                href='#'
                target='_blank'
                type='secondary'
                isDropdown
                isOpen={ false }
            />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});