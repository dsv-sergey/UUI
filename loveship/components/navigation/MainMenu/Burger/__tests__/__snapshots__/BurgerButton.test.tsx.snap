// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BurgerButton should be rendered correctly 1`] = `
<div
  className="container uui-button-box uui-enabled root button-primary indent-0"
  href={null}
  onClick={[Function]}
  onKeyDown={[Function]}
  tabIndex={0}
/>
`;

exports[`BurgerButton should be rendered correctly with props 1`] = `
<a
  className="container uui-button-box uui-enabled -clickable root button-secondary indent-0 dropdown hasIcon"
  href="#"
  onClick={[Function]}
  onKeyDown={[Function]}
  tabIndex={0}
  target="_blank"
>
  <div
    className="container uui-icon uui-enabled"
  >
    <svg
      className=""
      height={24}
      viewBox="0 0 12 24"
      width={12}
    >
      <use
        xlinkHref="#triangle.svg"
      />
    </svg>
  </div>
  <div
    className="container uui-icon uui-enabled"
  >
    <svg
      className=""
      height={24}
      viewBox="0 0 12 24"
      width={12}
    >
      <use
        xlinkHref="#accept-12.svg"
      />
    </svg>
  </div>
  <div
    className="uui-caption"
  >
    Test button
  </div>
</a>
`;
